service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/storageContainer/VerifyAzureStorageContainerCanBeCreatedStep.java

```diff
@@ -169,6 +169,8 @@ private StepResult validateLandingZoneSharedStorageAccountExist(
                   "Shared storage account not found in landing zone. Landing zone ID='%s'.",
                   landingZoneId)));
     } catch (IllegalStateException illegalStateException) { // Thrown by landingZoneApiDispatch
+      logger.error(
+          "Landing zone associated with the billing profile not found.", illegalStateException);
       return new StepResult(
```

Comment:
Can we not include the `illegalStateException` stack trace in the `LandingZoneNotFoundException` itself?

Reply:
Best practice is not to expose backtraces through the REST API. They can contain data that should not be visible to callers.

Not sure that applies in this case. I would expect LZ to be logging the backtrace before converting the exception.